tendrl/node_agent/node_sync/__init__.py

```diff
@@ -62,6 +62,17 @@ def _run(self):
                 tendrl_ns.node_agent.objects.NodeContext(tags=tags).save()
                 gevent.sleep(interval)
 
+                if tendrl_ns.tendrl_context.integration_id:
```

Comment:
You had to also check if the integration_id actually exists at /clusters/$integration_id, please do that